internal/mesh/internal/controllers/routes/generate.go

```diff
@@ -266,21 +277,37 @@ func compile(
 
 			svc := related.GetService(svcRef)
 			failoverPolicy := related.GetFailoverPolicyForService(svcRef)
-			destConfig := related.GetDestinationPolicyForService(svcRef)
+			destPolicy := related.GetDestinationPolicyForService(svcRef)
 
 			if svc == nil {
 				panic("impossible at this point; should already have been handled before getting here")
 			}
-			details.Service = svc.Data
+
+			// Find the destination proxy's port.
+			//
+			// Endpoints refs will need to route to mesh port instead of the
+			// destination port as that is the port of the destination's proxy.
+			//
+			// Note: we will always find a port here because we only add targets that have
+			// mesh ports above in shouldRouteTrafficToBackend().
+			for _, port := range svc.Data.Ports {
+				if port.Protocol == pbcatalog.Protocol_PROTOCOL_MESH {
+					details.MeshPort = port.TargetPort
```

Comment:
This now grabs the mesh port during ComputedRoutes construction so that the sidecar controller can skip resolving the Service resource just to get this data out again.